### PR TITLE
[Docs] Updated image-component example 

### PR DESCRIPTION
**Pull request by HaNdTriX**
## Changelog

- migrated example to typescript
- abstracted layout stuff to make example more readable
- adjust layout

## Documentation / Examples

- [x] Make sure the linting passes by running `pnpm lint`
- [x] The examples guidelines are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples)

### CODE REVIEW

File: examples/image-component/package.json [hunk 1]
```diff
@@ -9,5 +9,10 @@
     "next": "latest",
     "react": "^17.0.2",
     "react-dom": "^17.0.2"
```
Comment (balazsorban44) [inline examples/image-component/package.json:11]: Could we bump these?
```suggestion
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
```

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Thanks, added some comments. Could you address those?

**Comment by HaNdTriX** [comment]
@balazsorban44 Thanks for your review!

File: examples/image-component/styles/globals.css [hunk 2]
```diff
@@ -6,6 +6,14 @@ body {
   color: white;
 }
 
+#__next {
+  padding: 4rem 1rem;
+}
+
+#__next p {
+  margin: 1.5rem 0;
+}
```
Comment (styfle) [inline examples/image-component/styles/globals.css:15]: These don't seem specific to typescript

**Review by styfle — CHANGES_REQUESTED** [review]
This breaks the background image demo 

I suggest creating the first PR that is just JS => TS. 

The create a second PR that makes style changes.

image

**Comment by HaNdTriX** [comment]
> This breaks the background image demo

Good catch! I have fixed this issue. Next time I will keep the git history clean I promise 😜.

**Comment by HaNdTriX** [comment]
closing in favour of https://github.com/vercel/next.js/pull/40204

**Pull request closed without merging**